examples/examples/compose_extrinsic_no_std.rs

```diff
@@ -0,0 +1,95 @@
+
+use kitchensink_runtime::{BalancesCall, Runtime, RuntimeCall, Signature};
+use sp_keyring::AccountKeyring;
+use sp_runtime::{generic::Era, MultiAddress};
+use substrate_api_client::{
+	ac_compose_macros::compose_extrinsic_offline,
+	ac_primitives::{
+		extrinsic_params::AssetBalanceFor, AssetTip, AssetTipExtrinsicParams, ExtrinsicParams,
+		ExtrinsicSigner, FrameSystemConfig, GenericAdditionalParams, GenericExtrinsicParams,
+	},
+	rpc::JsonrpseeClient,
+	Api, GetChainInfo, SubmitExtrinsic,
+};
+
+type Hash = <Runtime as FrameSystemConfig>::Hash;
+
+#[tokio::main]
+async fn main() {
+	env_logger::init();
+
+	// Initialize api and set the signer (sender) that is used to sign the extrinsics.
+	let signer = AccountKeyring::Alice.pair();
+	let client = JsonrpseeClient::with_default_url().unwrap();
+	// Api::new(..) is not actually an offline call, but retrieves metadata and other information from the node.
+	// If this is not acceptable, use the Api::new_offline(..) function instead. There are no examples for this,
+	// because of the constantly changing substrate node. But check out our unit tests - there are Apis created with `new_offline`.
+	//
+	// ! Careful: AssetTipExtrinsicParams is used here, because the substrate kitchensink runtime uses assets as tips. But for most
+	// runtimes, the PlainTipExtrinsicParams needs to be used.
+	let mut api = Api::<
+		ExtrinsicSigner<sp_core::sr25519::Pair, Signature, Runtime>,
+		JsonrpseeClient,
+		AssetTipExtrinsicParams<Runtime>,
+		Runtime,
+	>::new(client.clone())
+	.unwrap();
+	let extrinsic_signer =
+		ExtrinsicSigner::<sp_core::sr25519::Pair, Signature, Runtime>::new(signer);
+	let signer_clone = extrinsic_signer.clone();
+	// Signer is needed to get the nonce
+	api.set_signer(signer_clone);
+
+	// Information for Era for mortal transactions (online).
+	let last_finalized_header_hash = api.get_finalized_head().unwrap().unwrap();
+	let header = api.get_header(Some(last_finalized_header_hash)).unwrap().unwrap();
+	let period = 5;
+
+	// Get information out of Api (online).
+	let spec_version = api.runtime_version().spec_version;
+	let transaction_version = api.runtime_version().transaction_version;
+	let genesis_hash = api.genesis_hash();
+	let signer_nonce = api.get_nonce().unwrap();
+	println!("[+] Alice's Account Nonce is {}\n", signer_nonce);
+	let recipient = MultiAddress::Id(AccountKeyring::Bob.to_account_id());
+
+	// Construct extrinsic without using Api (no_std).
+	let additional_extrinsic_params: GenericAdditionalParams<
+		AssetTip<AssetBalanceFor<Runtime>>,
+		Hash,
+	> = GenericAdditionalParams::new()
+		.era(Era::mortal(period, header.number.into()), last_finalized_header_hash)
+		.tip(0);
+	let extrinsic_params = GenericExtrinsicParams::new(
+		spec_version,
+		transaction_version,
+		signer_nonce,
+		genesis_hash,
+		additional_extrinsic_params,
+	);
+	let call =
+		RuntimeCall::Balances(BalancesCall::transfer_allow_death { dest: recipient, value: 42 });
```

Comment:
Same here for the `RuntimeCall`: In no_std the runtime compiles to wasm, so tricky to import in no_std environments. Better to go without it. That's the fun part now: To get the Call you need the metadata.

Check out how it's done in the worker:
metadata call: https://github.com/integritee-network/worker/blob/master/core-primitives/node-api/metadata/src/lib.rs#L72-L88
Retrieve indexes for a specific pallet: https://github.com/integritee-network/worker/blob/master/core-primitives/node-api/metadata/src/pallet_teeracle.rs